File: src/console/command/impl/DebugCommand.php
<?php

namespace pocketcloud\cloud\console\command\impl;

use pocketcloud\cloud\config\impl\LogSettingsConfig;
use pocketcloud\cloud\config\impl\MainConfig;
use pocketcloud\cloud\console\command\Command;
use pocketcloud\cloud\console\command\sender\ICommandSender;
use pocketcloud\cloud\console\command\SubCommand;
use pocketcloud\cloud\console\handler\ExceptionHandler;

final class DebugCommand extends Command {

    public function __construct() {
        parent::__construct("debug", "Enable or disable the debug mode", ["deb"]);
    }

    public function run(ICommandSender $sender, string $label, array $args, ?SubCommand $subCommand = null): bool {
        if (LogSettingsConfig::getInstance()->isDebugMode()) {
            $sender->success("The §edebug mode §rhas been §cdisabled§r!");
            LogSettingsConfig::getInstance()->setDebugMode(false);
        } else {
            $sender->success("The §edebug mode §rhas been §aenabled§r!");
            LogSettingsConfig::getInstance()->setDebugMode(true);
        }

        ExceptionHandler::tryCatch(fn() => MainConfig::getInstance()->save(), "Failed to save main config");
        return true;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/impl/DebugCommand.java
package de.pocketcloud.cloud.console.command.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.config.LogSettingsConfig;
import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.ctx.CommandContext;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.sender.CommandSender;
import de.pocketcloud.cloud.console.log.CloudLogger;

@CommandDescription(name = "debug", description = "Toggle the debug mode")
public final class DebugCommand extends Command {

    @Override
    public void prepare() {}

    @Override
    public boolean run(CommandSender sender, CommandContext ctx) {
        PocketCloud.instance().logSettingsConfig().setDebugMode(!PocketCloud.instance().logSettingsConfig().isDebugMode());
        PocketCloud.instance().logSettingsConfig().save();
        CloudLogger.get().setDebugMode(PocketCloud.instance().logSettingsConfig().isDebugMode());

        if (PocketCloud.instance().logSettingsConfig().isDebugMode()) {
            sender.success("Successfully §aenabled §rthe §6debug mode§r.");
        } else {
            sender.success("Successfully §cdisabled §rthe §6debug mode§r.");
        }

        return true;
    }
}